### PR TITLE
Check persistency of extensions before insert

### PR DESCRIPTION
**Pull request by etiennehomer**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)

**What kind of change does this PR introduce?**
Extension can now be persistent or not. This permit to avoid persisting private extensions that are only used during the import.

### CODE REVIEW

**Review by antoinebhs — APPROVED** [review]
Maybe you'll have to add a test for the coverage?

**Review by Mathieu-Deharbe — APPROVED** [review]

File: network-store-server/src/test/java/com/powsybl/network/store/server/ExtensionHandlerTest.java [hunk 1]
```diff
@@ -289,4 +290,38 @@ public void updateExtensionsFromEquipmentsTest() {
         assertEquals(10.0, activePowerControl.getDroop(), 0.1);
         assertEquals(2.0, activePowerControl.getParticipationFactor(), 0.1);
     }
+
+    @Test
+    public void insertNonPersistentExtensionTest() {
+        String equipmentIdA = "idBatteryA";
+
+        OwnerInfo infoBatteryA = new OwnerInfo(
```
Comment (Mathieu-Deharbe) [inline network-store-server/src/test/java/com/powsybl/network/store/server/ExtensionHandlerTest.java:298]: Absolutely unimportant but I find it strange to call it sometimes equipmentA and sometimes BatteryA.

Actually even the "A" is suspicisous, when there is only one equipment. It looks like a copy paste!

File: network-store-server/src/test/java/com/powsybl/network/store/server/ExtensionHandlerTest.java [hunk 2]
```diff
@@ -289,4 +290,38 @@ public void updateExtensionsFromEquipmentsTest() {
...
+                NETWORK_UUID,
+                Resource.INITIAL_VARIANT_NUM
+        );
+        Map<String, ExtensionAttributes> extensionAttributesMapA = Map.of(
+                "notPersistent", new NonPersistentExtensionAttributes(),
+                "activePowerControl", ActivePowerControlAttributes.builder().droop(6.0).participate(true).participationFactor(1.5).build()
+        );
+        Map<OwnerInfo, Map<String, ExtensionAttributes>> mapA = new HashMap<>();
+        mapA.put(infoBatteryA, extensionAttributesMapA);
+        extensionHandler.insertExtensions(mapA);
```
Comment (Mathieu-Deharbe) [inline network-store-server/src/test/java/com/powsybl/network/store/server/ExtensionHandlerTest.java:310]: `extensionHandler.insertExtensions(Map.of(infoBattery, extensionAttributes));`

**Review by Mathieu-Deharbe — APPROVED** [review]
Test ok, only small remarks.

File: network-store-server/src/test/java/com/powsybl/network/store/server/ExtensionHandlerTest.java (hunk 1 above)
Comment (etiennehomer) [reply network-store-server/src/test/java/com/powsybl/network/store/server/ExtensionHandlerTest.java:298]: It was copied from another test

File: network-store-server/src/test/java/com/powsybl/network/store/server/ExtensionHandlerTest.java (hunk 2 above)
Comment (etiennehomer) [reply network-store-server/src/test/java/com/powsybl/network/store/server/ExtensionHandlerTest.java:310]: done

**Comment by etiennehomer** [comment]
> Maybe you'll have to add a test for the coverage?

I added a test

**Review by antoinebhs — APPROVED** [review]

**Review by antoinebhs — APPROVED** [review]

**Pull request merged**